tensorboard/plugins/mesh/metadata_test.py

```diff
@@ -93,6 +101,16 @@ def test_metadata_format(self):
     with six.assertRaisesRegex(self, TypeError, r'Content type must be bytes.'):
       metadata.parse_plugin_metadata(123)
 
+  def test_default_components(self):
+    """Tests that defult components are added when necessary."""
+    self._create_metadata()
+    result = FakeResult(metadata.get_current_version(), 0)
+    with patch.object(
+        plugin_data_pb2.MeshPluginData, 'FromString',
```

Comment:
No need to mock here at all—`parse_plugin_metadata` is a pure function;
you can just pass it the appropriate input:

```python
result = plugin_data_pb2.MeshPluginData(
    version=metadata.get_current_version(),
    components=0,
)
metadata.parse_plugin_metadata(result.SerializeToString())
```